### DISCUSSION TITLE
xml object with no parent node

### DISCUSSION THREAD

**Discussion by 10062244** [FreeOpcUa/opcua-asyncio#1000] [Q&A]
Hello,
Ive been working with the euromap 82.1 spec trying to create a server in a temp control device. I import all of the required xml files and Instantiate temp control device object using the TCD_Interface_Type but no matter what I do I cant get the client in the mold machine to see this object. It connects to the sever, but will not recognize my TCD node. I can see it with UAExpert and everything seems to be ok.

![image](https://user-images.githubusercontent.com/26574987/185208709-444dd28a-ad58-438a-bd2f-96a7e1c9ff39.png)

The only issue I can identify is that in order to get the Opc.Ua.PlasticsRubber.GeneralTypes.NodeSet2.xml file to import I had to add add ParentNodeId entries

Original on the right
![image](https://user-images.githubusercontent.com/26574987/185210370-4b6c231b-3b28-4131-9ffb-ec98b1677d12.png)

or the importer would give me a bad parent node id error. Should an object be able to be imported with out a parent node id? or is this an error in the spec file? Ill attach the files im trying to import.

Back to the original issue. I know this may be a little out of scope, but should instantiated objects be in a specific namespace? The 82.1 spec seems to indicate that object instances should be in ns index 1. The spec comes with a csv file that includes symbols, identifier numbers, and types. Should I use this file when instantiating my object some how? I am wondering if my client is not seeing my object because the node identifier numbers are incorrect. 

any input would be greatly appreciated.
Thanks

[xml_specs.zip](https://github.com/FreeOpcUa/opcua-asyncio/files/9365157/xml_specs.zip)

**Comment by schroeder-** [comment]
Sry for the late response. The missing parent thing is as of the xsd illegal, but a lot of nodeset generator leave them out on DefaultXML and DefaultBinary nodes for datatypes. So we need to prepare our import to handle this corner cases as well. Is on my list for future fixes.
